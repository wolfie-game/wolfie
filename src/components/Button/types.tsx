import {MouseEventHandler} from 'react'

export type ButtonProps = {
  styleName?: string
  type: 'button' | 'submit' | 'reset' | undefined
  handler: MouseEventHandler
  children?: string
}
