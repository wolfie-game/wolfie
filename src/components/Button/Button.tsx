import React from 'react'
import { ButtonProps } from './types'

type Props = ButtonProps

const Button = (props: Props) => {
  return (
    <button className={props.styleName} type={props.type} onClick={props.handler}>{props.children}</button>
  )
}

export default Button