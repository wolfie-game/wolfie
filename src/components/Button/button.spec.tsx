import React from 'react'
import Button from './Button'
import renderer from 'react-test-renderer'

it('render correctly Input component', () => {
  const ButtonComponent = renderer
    .create(<Button type="button" handler={() => {}}></Button>)
    .toJSON()
  expect(ButtonComponent).toMatchSnapshot()
})
