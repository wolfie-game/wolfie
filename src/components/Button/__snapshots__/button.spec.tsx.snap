// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Input component 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;
